consumer/consumererror/error.go

```diff
@@ -0,0 +1,136 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package consumererror // import "go.opentelemetry.io/collector/consumer/consumererror"
+
+import (
+	"errors"
+	"net/http"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"go.opentelemetry.io/collector/consumer/consumererror/internal/statusconversion"
+)
+
+// Error is intended to be used to encapsulate various information that can add
+// context to an error that occurred within a pipeline component. Error objects
+// are constructed through calling `New` with the relevant options to capture
+// data around the error that occurred.
+//
+// Error should be obtained from a given `error` object using `errors.As`.
+type Error struct {
+	error
+	httpStatus  int
+	grpcStatus  *status.Status
+	isRetryable bool
+}
+
+var _ error = (*Error)(nil)
+
+// NewOTLPHTTPError records an HTTP status code that was received from a server
+// during data submission.
+func NewOTLPHTTPError(origErr error, httpStatus int) error {
```

Comment:
What is the expectation if calling this with `origErr == nil` and `httpStatus == 200`? If not expected at least document if not something else.